tests/cmdline/params/options/test_verbosity.py

```diff
@@ -0,0 +1,106 @@
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""Tests for the :class:`~aiida.cmdline.params.options.main.VERBOSITY` option."""
+import logging
+
+import click
+import pytest
+
+from aiida.cmdline.utils import echo
+from aiida.cmdline.params import options
+from aiida.common.log import AIIDA_LOGGER, LOG_LEVELS, configure_logging
+
+
+@click.command()
+@options.VERBOSITY()
+def cmd():
+    """Test command prints messages through the ``AIIDA_LOGGER`` and the ``CMDLINE_LOGGER``.
+
+    The messages to the ``CMDLINE_LOGGER`` are performed indirect through the utilities of the ``echo`` module.
+    """
+    for log_level in LOG_LEVELS.values():
+        AIIDA_LOGGER.log(log_level, 'aiida')
+
+    echo.echo_debug('verdi')
+    echo.echo_info('verdi')
+    echo.echo_report('verdi')
+    echo.echo_warning('verdi')
+    echo.echo_error('verdi')
+    echo.echo_critical('verdi')
+
+
+@pytest.mark.usefixtures('reset_log_level')
+def test_default(run_cli_command):
+    """Test the command without explicitly specifying the verbosity.
+
+    The default log level is ``REPORT`` so its messages and everything above should show and the rest not.
+    """
+    result = run_cli_command(cmd, raises=True)
+
+    for log_level_name, log_level in LOG_LEVELS.items():
+        if log_level >= logging.REPORT:  # pylint: disable=no-member
+            assert f'{log_level_name.capitalize()}: verdi' in result.output
+            assert f'{log_level_name.capitalize()}: aiida' in result.output
+        else:
+            assert f'{log_level_name.capitalize()}: verdi' not in result.output
+            assert f'{log_level_name.capitalize()}: aiida' not in result.output
+
+
+@pytest.mark.parametrize('option_log_level', [level for level in LOG_LEVELS.values() if level != logging.NOTSET])
+@pytest.mark.usefixtures('reset_log_level')
+def test_explicit(run_cli_command, option_log_level):
+    """Test explicitly settings a verbosity"""
+    log_level_name = logging.getLevelName(option_log_level)
+    result = run_cli_command(cmd, ['--verbosity', log_level_name], raises=True)
+
+    for log_level_name, log_level in LOG_LEVELS.items():
+        if log_level >= option_log_level:
+            assert f'{log_level_name.capitalize()}: verdi' in result.output
+            assert f'{log_level_name.capitalize()}: aiida' in result.output
+        else:
+            assert f'{log_level_name.capitalize()}: verdi' not in result.output
+            assert f'{log_level_name.capitalize()}: aiida' not in result.output
+
+
+@pytest.mark.usefixtures('reset_log_level')
+def test_config_aiida_loglevel(run_cli_command, isolated_config, caplog):
+    """Test the behavior of the ``--verbosity`` option when the ``logging.aiida_loglevel`` config option is set.
+
+    Even though the ``CMDLINE_LOGGER`` is technically a child of the ``AIIDA_LOGLEVEL`` and so normally the former
```

Comment:
Since, as you mention, this is a bit counterintuitive behavior, is there something that speaks against reversing the inheritance and making the `CMDLINE_LOGGER` the parent of the AiiDA logger?

In principle one could then even decide not to care about profile-specific reconfiguration of the AiiDA logger (but I guess it's probably still wise to enforce the log level handed down from the command line also there).

Reply:
We could try, but I think logger inheritance in the `logging` module works by the module path. So the `aiida` module has `aiida` as the logger name and the `aiida.cmdline` module has `aiida.cmdline` and therefore is a child. I am not even sure if we can even make a logger the parent of `aiida` since that is a top-level namespace.

Reply:
> We could try, but I think logger inheritance in the `logging` module works by the module path.

I think that is what happens if you do `logging.getLogger(__name__)` because `logging.getLogger('aiida.submodule')` means "create logger `submodule` as child of logger `aiida`)
See the [logging docs](https://docs.python.org/3/library/logging.html#logger-objects).

So one could in principle do something like
```
CMDLINE_LOGGER = logging.getLogger('aiida_cli')  # use some unique name
AIIDA_LOGGER = CMDLINE_LOGGER.get('aiida')
```
but that would change the name of the aiida logger to 'aiida_cli.aiida' which does not follow standard practices.

We can do something else though:
```
CMDLINE_LOGGER = logging.getLogger('aiida')
AIIDA_LOGGER = CMDLINE_LOGGER.getChild('api')
```
and let `configure_logging` configure the `aiida.api` logger.

Anyhow, it may be somewhat confusing either way, so I leave the decision up to you.

Reply:
That's also what I surmised. Given that it anyway would be not ideal and there would always be something that would be an bit backwards, I would keep what I have currently.